### PR TITLE
feat: implement customer and account creation

### LINKED ISSUE #13: Create an account
*As* bank employe
*I want* to create an account for a customer
*so* that the customer can transfer money over the account and provide us with fresh, juicy money.

### Creating an account
**Given** the customer doesn't already have an account
**When** I create an account via the customers customer number
**The** account was created
**And** was propagated via an AccountCreatedEvent

### Don't create additional accounts
**Given** the customer already has an account
**When** I create an account via the customers customer number
**The** no account was created
**And** the existing account exists unchanged

### Don't create an account for a no
**Given** a non-existing customer number
**When** I create an account via the customers customer number
**The** no account was created

### PR DESCRIPTION
**Pull request by mmiikkkkaa**

### CODE REVIEW

File: demo-transactions/src/main/kotlin/io/holixon/cqrshexagonaldemo/demoparent/transactions/application/CreateAccountUsecase.kt [hunk 1]
```diff
@@ -3,17 +3,42 @@ package io.holixon.cqrshexagonaldemo.demoparent.transactions.application
...
 import io.holixon.cqrshexagonaldemo.demoparent.transactions.application.port.outbound.account.AccountOutPort
 import io.holixon.cqrshexagonaldemo.demoparent.transactions.application.port.outbound.customer.CustomerOutPort
-import io.holixon.cqrshexagonaldemo.demoparent.transactions.domain.Account
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.domain.event.AccountCreatedEvent
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.domain.model.account.Account
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.domain.model.customer.CustomerNumber
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.domain.service.CustomerAccountVerificationService
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.domain.service.IbanCreationService
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Component
```
Comment (janweinschenker) [inline demo-transactions/src/main/kotlin/io/holixon/cqrshexagonaldemo/demoparent/transactions/application/CreateAccountUsecase.kt]: In the application package, classes should not import framework code.

I would not use `@Component` or the `ApplicationEventPublisher`

For instantiating beans, I would have used the Beans Definition DSL (see `CommandApplication`). And I would have written an adapter to publish events

**Review by janweinschenker — APPROVED** [review]
As discussed, the `@Component`annotation can be removed and the event publisher could be wrappen in an adapter.

**Comment by rafaelcp94** [comment]
Should we take advantage of extension functions to build our entities or DTO's ? i see some advantages to do that approach because we have cleaner Code, readability. On the other hand with class component we have better isolation flexibility. Do you think its worth to discuss this ?

**Comment by mmiikkkkaa** [comment]
> Should we take advantage of extension functions to build our entities or DTO's ? i see some advantages to do that approach because we have cleaner Code, readability. On the other hand with class component we have better isolation flexibility. Do you think its worth to discuss this ?

Its always worth discussing, but I need it a little bit more precise: in what spots would you like to use extension functions?

**Pull request merged**